llvm/lib/Transforms/Scalar/InferAddressSpaces.cpp

```diff
@@ -1311,7 +1311,9 @@ bool InferAddressSpacesImpl::rewriteWithNewAddressSpaces(
 
           while (isa<PHINode>(InsertPos))
             ++InsertPos;
-          U.set(new AddrSpaceCastInst(NewV, V->getType(), "", &*InsertPos));
+          // This instruction may contain multiple uses of V, update them all.
```

Comment:
The `skipToNextUser` on line 1227 means that this for loop runs once for each `User`, not once for each `Use`, so every instance of `U.set` seems wrong, since it will leave other uses in the same instruction untouched.